### PR TITLE
Fix user schema for testresults

### PR DESCRIPTION
**Pull request by kaipot**
#### Rationale
Update to fix testresults schema, change DeleteRunAction and FlagRunAction to inherit from MutatingApiAction

### CODE REVIEW

File: testresults/src/org/labkey/testresults/TestResultsController.java [hunk 1]
```diff
@@ -687,22 +688,23 @@ public ModelAndView getView(Object o, BindException errors) throws Exception
                 Table.delete(TestResultsSchema.getTableInfoHangs(), filter);
                 Table.delete(TestResultsSchema.getTableInfoTestRuns(), rowId); // delete run last because of foreign key
                 transaction.commit();
+            } catch (Exception x) {
+                response.put("success", false);
```
Comment (labkey-jeckels) [inline testresults/src/org/labkey/testresults/TestResultsController.java:692]: Logging this on the server side can sometimes be helpful to capture the full error info, including its stack trace. But this really depends on what types of exceptions you might expect

**Review by labkey-jeckels — APPROVED** [review]
A small, optional suggestion

File: testresults/src/org/labkey/testresults/view/runDetail.jsp [hunk 2]
```diff
@@ -68,8 +68,7 @@
     <!--Header content, username run Id, and other information about the run-->
     <h2>
         <%=h(run.getUserName() + " : " + df.format(run.getPostTime()))%> &nbsp;
-        <img id="flagged" flagged="false"
-            src="<%=h(contextPath)%>/TestResults/img/<% if (run.isFlagged()) { %>flagon<% } else { %>flagoff<% } %>.png"
+        <img id="flagged" src="<%=h(contextPath)%>/TestResults/img/<% if (run.isFlagged()) { %>flagon<% } else { %>flagoff<% } %>.png"
```
Comment (vagisha) [inline testresults/src/org/labkey/testresults/view/runDetail.jsp]: Maybe replace this with
src="<%=getWebappURL("/TestResults/img/" + (run.isFlagged() ? "flagon" : "flagoff" + ".png"))%>"

**Review by vagisha — APPROVED** [review]

**Comment by labkey-tchad** [comment]
@labkey-jeckels @vagisha @kaipot 
I think the `testresults.xml` was fine the way it was. It is failing the [SchemaXMLTest now](https://teamcity.labkey.org/buildConfiguration/LabKey_217Release_Community_DrtPostgres/1732232?buildTab=tests&expandedTest=build%3A%28id%3A1732232%29%2Cid%3A4516).
We noticed the error when merging 21.7 to 21.11 today.

The [upgrade script](https://github.com/LabKey/MacCossLabModules/blob/develop/testresults/resources/schemas/dbscripts/postgresql/testresults-13.29-13.30.sql) that moved the `active` column from `user` to `userdata` is commented out.

**Comment by labkey-jeckels** [comment]
@labkey-tchad @kaipot @labkey-sweta as far as TeamCity goes, that's correct. However, looking at skyline.ms (the server where this module is used), I believe that the script was probably run manually against the DB. It doesn't have a testresults.user.active column and does have a testresults.userdata.active column.

Thus, I suggest that we fix the tests by uncommenting the script. This won't affect skyline.ms as its schema version is already at 13.38.

**Comment by vagisha** [comment]
Thanks, Josh.  Yes, that script must have been run manually on skyline.ms. Going forward, we will always use the installer builds from TeamCity for upgrading skyline.ms so that we don't see such discrepancies.

**Comment by labkey-jeckels** [comment]
https://github.com/LabKey/MacCossLabModules/pull/178

**Pull request merged**